we.go

```diff
@@ -403,22 +401,26 @@ func formatWind(c cond) string {
 }
 
 func formatVisibility(c cond) string {
-	distUnit := float32(c.VisibleDistKM)
 	if config.Imperial {
-		distUnit = float32(c.VisibleDistKM) * 0.621
+		c.VisibleDistKM = (c.VisibleDistKM * 621) / 1000
 	}
-	return fmt.Sprintf("%d %s            ", int32(distUnit), unitVis[config.Imperial])[:15]
+	return fmt.Sprintf("%d %s            ", c.VisibleDistKM, unitVis[config.Imperial])[:15]
 }
 
 func formatRain(c cond) string {
-	rainUnit := float32(c.PrecipMM)
 	if config.Imperial {
-		rainUnit = float32(c.PrecipMM) * 0.039
+		temp := c.PrecipMM * 39 
+		c.PrecipMM = temp / 1000 // Temperature in Fahrenheit
+		decimal := temp % 10     // Record one decimal point
```

Comment:
The variable should be called `tmp` instead of `temp` to not confuse it with temperature. The comment is wrong of course, since you are converting distance units (mm => in).

Also the calculation of the decimal point value is wrong. It should be `decimac := (temp / 100) % 10`.